### PR TITLE
Add support for Ubuntu 24

### LINKED ISSUE #270: Ubuntu 24.04 support
GitHub has recently released Ubuntu 24.04 support in beta on GitHub actions:

https://github.blog/changelog/2024-05-14-github-hosted-runners-public-beta-of-ubuntu-24-04-is-now-available/

Could support for this be added? Currently using the beta `ubuntu-24.04` runner gives:

```
Tried to map a target OS from env. variable 'ImageOS' (got ubuntu24), but failed. If you're using a self-hosted runner, you should set 'env': 'ImageOS': ... to one of the following: ['ubuntu18', 'ubuntu20', 'ubuntu22', 'win19', 'win22']
```

Thanks!

### PR DESCRIPTION
**Pull request by sax**
# Description

As of right now, the only packages available for OTP 27 are only available for Ubuntu 24.04. The setup-beam action only supports Ubuntu 22 or earlier.

These changes seem to work for me, though I am currently unable to run the tests... will add more info in the comments.

Closes #270.

- [x] I have performed a self-review of my changes
- [x] I have read and understood the [contributing guidelines](/erlef/setup-beam/blob/main/CONTRIBUTING.md)

### CODE REVIEW

**Comment by sax** [comment]
When I try to run the tests locally, I get the following error:

```
::error::Error: Got an exception when trying to parse non-JSON list {"message":"Bad credentials","documentation_url":"https://docs.github.com/rest"}: Error: expected a list!
```

I don't see any information in the README or in the code about how to set up the project for running tests locally, but am happy to do so if someone can point me in a good direction.

**Comment by ericmj** [comment]
> When I try to run the tests locally, I get the following error

You may have to generate a github access token [1] and set `GITHUB_TOKEN`.

[1] https://github.com/settings/tokens

**Comment by ericmj** [comment]
Can you add ubuntu-24.04 to the matrix here? https://github.com/erlef/setup-beam/blob/main/.github/workflows/ubuntu.yml

ubuntu-latest currently points to ubuntu-22.04.

@paulo-ferraz-oliveira Should we be explicitly with the ubuntu version instead of using latest?

**Comment by sax** [comment]
Awesome, `GITHUB_TOKEN` was the thing I was missing. I may issue another pull request adding a reference to that in the README.

@ericmj One thing that I ran into today is that there are no Elixir builds yet for OTP 27 in [builds.hex.pm](https://builds.hex.pm/builds/elixir/builds.txt). Do you have a preference for how many OTP versions (and which versions) I add to the matrix?

**Comment by ericmj** [comment]
> One thing that I ran into today is that there are no Elixir builds yet for OTP 27 in [builds.hex.pm](https://builds.hex.pm/builds/elixir/builds.txt).

The builds were delayed but are available now.

> Do you have a preference for how many OTP versions (and which versions) I add to the matrix?

No, I would defer to @paulo-ferraz-oliveira here.

**Comment by sax** [comment]
Ok, based on the CI failures and looking deeper at the test suite I just reran the tests locally in an Ubuntu container in docker.

**Comment by sax** [comment]
Also to be very transparent I'm totally happy with anybody updating these changes, or throwing them away in favor of something else if that's decided to be better. Attribution is nice, but being able to update my apps to OTP 27 would be _very_ nice 😄.

File: .github/workflows/ubuntu.yml [hunk 1]
```diff
@@ -39,6 +39,14 @@ jobs:
             elixir-version: '1.14.3'
             os: 'ubuntu-latest'
             version-type: 'strict'
+          - otp-version: '27.0'
+            os: 'ubuntu-24.04'
+            version-type: 'strict'
+          - otp-version: '26.0'
+            os: 'ubuntu-24.04'
+            version-type: 'strict'
+          - otp-version: '25.0'
```
Comment (paulo-ferraz-oliveira) [inline .github/workflows/ubuntu.yml:48]:
```suggestion
          - otp-version: '25'
```

File: .github/workflows/ubuntu.yml [hunk 2]
```diff
@@ -39,6 +39,14 @@ jobs:
             elixir-version: '1.14.3'
             os: 'ubuntu-latest'
             version-type: 'strict'
+          - otp-version: '27.0'
+            os: 'ubuntu-24.04'
+            version-type: 'strict'
+          - otp-version: '26.0'
```
Comment (paulo-ferraz-oliveira) [inline .github/workflows/ubuntu.yml:45]:
```suggestion
          - otp-version: '26'
```

File: .github/workflows/ubuntu.yml [hunk 3]
```diff
@@ -39,6 +39,14 @@ jobs:
             elixir-version: '1.14.3'
             os: 'ubuntu-latest'
             version-type: 'strict'
+          - otp-version: '27.0'
```
Comment (paulo-ferraz-oliveira) [inline .github/workflows/ubuntu.yml:42]:
```suggestion
          - otp-version: '27'
```

Comment (paulo-ferraz-oliveira) [reply .github/workflows/ubuntu.yml:42]: This way we make it choose the latest `27.x.y.z...`, not the latest `27.0.y.z...` 😄

**Comment by paulo-ferraz-oliveira** [comment]
The proposed changes seem sane to me.

> the only packages available for OTP 27 are only available for Ubuntu 24.04

What does this mean? From `builds.txt`? I think in the meantime more stuff is available, if that's the case.

**Review by paulo-ferraz-oliveira — APPROVED** [review]
Feel free to merge with or without the code change suggestions.

**Comment by sax** [comment]
Cool! Decided to apply your suggestions as a single commit, vs 3 commits via the GitHub UI.

> the only packages available for OTP 27 are only available for Ubuntu 24.04

Yeah looks like this was just an issue yesterday, and built are coming out for the different platforms. So not a real issue, just an explanation of why I chose to open this PR yesterday.

File: .github/workflows/ubuntu.yml (hunk 1 above)
Comment (sax) [reply .github/workflows/ubuntu.yml]: Applied in https://github.com/erlef/setup-beam/pull/271/commits/d2ad6440a8f7071bde167f803e52e9a9a9df10e8

File: .github/workflows/ubuntu.yml (hunk 2 above)
Comment (sax) [reply .github/workflows/ubuntu.yml]: Applied in https://github.com/erlef/setup-beam/pull/271/commits/d2ad6440a8f7071bde167f803e52e9a9a9df10e8

File: .github/workflows/ubuntu.yml (hunk 3 above)
Comment (sax) [reply .github/workflows/ubuntu.yml]: Applied in https://github.com/erlef/setup-beam/pull/271/commits/d2ad6440a8f7071bde167f803e52e9a9a9df10e8

**Comment by paulo-ferraz-oliveira** [comment]
> So not a real issue, just an explanation of why I chose to open this PR yesterday.

No prob. Now we're `ubuntu24` ready, too 🤗. I'm gonna do some FOSS related to OTP 27 today, still, and it's possible I release the action's new version later (today, still). If I don't kindly remind my by commenting on this pr.

**Comment by paulo-ferraz-oliveira** [comment]
Thanks for the PR.

**Pull request merged**